src/types/index.ts

```diff
@@ -1,6 +1,74 @@
 import type { MENU_LIST, STATION_DETAILS_CATEGORIES } from '@constant';
 
-export interface StationProps {
+export interface ColumnType<T, K extends keyof T> {
+  key: K;
+  header: string;
+  width?: number;
+}
+
+interface FaultReportDetails {
+  id: number;
+  stationId: string;
+  memberId: number;
+}
+
+interface MemberDetails {
+  id: number;
+  email: number;
+  role: string;
+}
+
+const memberColumns: ColumnType<MemberDetails, keyof MemberDetails>[] = [
+  {
+    key: 'id',
+    header: 'ID',
+  },
+  {
+    key: 'email',
+    header: 'E-MAIL',
+  },
+  {
+    key: 'role',
+    header: '회원 권한',
+  },
+];
+const faultReportColumns: ColumnType<FaultReportDetails, keyof FaultReportDetails>[] = [
+  {
+    key: 'id',
+    header: 'ID',
+  },
+  {
+    key: 'stationId',
+    header: '충전소 ID',
+  },
+  {
+    key: 'memberId',
+    header: '회원 ID',
+  },
+];
```

Comment:
이렇게 만든 이유가 있을까요??

```suggestion
const faultReportColumns = {
 id: 'ID',
 stationId: '충전소 ID'
 ....
}

const faultRepoutColumnList = getTypedObjectEntries(faultReportColumns) 
```

이렇게 만들어서 쓰면 안되나요??

Reply:
저는 혹시 key, header 외의 다른 속성들을 넣을 수 있지 않을까라는 생각에 했습니다. 하지만 굳이 지금 단계에선 과하고 코드의 복잡도만 높아지겠네요 수정하겠습니다